k8s/ci/test-script.sh

```diff
@@ -554,3 +554,149 @@ testPoolReadOnly() {
+            break
+        fi
+        sleep 5
+    done
+
+    if [ $success == 0 ]; then
+        echo "Failed to execute the command $command"
+        exit 1
+    fi
+    echo "Command $command executed successfully"
+}
+
+
+echo "===========Testing Snapshots On CVR By Enabling Feature Gate On CStor Pools ============="
+## Get the deployment name of CSP
+pool_dep_list=( $(kubectl get deployment -l app=cstor-pool -o jsonpath='{.items[?(@.metadata.labels.openebs\.io/storage-pool-claim=="sparse-claim-auto")].metadata.name}' -n openebs))
+pool_dep=${pool_dep_list[0]}
+
+## Enable the feature gates by patching the deployment with corresponding feature gates
+## NOTE: If deployment already patched then exit code will be 0
+kubectl patch deployment --namespace openebs ${pool_dep} --patch='{"spec": {"template": {"spec": {"containers": [{"name": "cstor-pool-mgmt","env": [{"name": "REBUILD_ESTIMATES", "value": "true"}]}]}}}}'
+rc=$?
+if [ $rc != 0 ]; then
+    echo "Failed to patch ${pool_dep} deployment to enable REBUILD_ESTIMATE feature gates"
+    exit 1
+fi
+
+## If Deployment patched checking the rollout status
+rollout_status=$(kubectl rollout status --namespace openebs deployment/$pool_dep)
+rc=$?; if [[ ($rc -ne 0) || ! (${rollout_status} =~ "successfully rolled out") ]];
+    then echo "ERROR: Failed to rollout status for $pool_dep error: $rc"; exit; fi
+
+## Get volume snapshot data name
+volume_snapshot_data_name=$(kubectl get volumesnapshot snapshot-demo-cstor -ojsonpath='{.spec.snapshotDataName}')
+rc=$?
+if [ $rc != 0 ]; then
+    echo "Failed to get volumesnapshotdata name for volumesnapshot: ${volumeSnapshotDataName}"
+    exit 1
+fi
+
+## Get Snapshot name from volume snapshot data
+k8s_snapshot_name=$(kubectl get volumesnapshotdata ${volume_snapshot_data_name} -ojsonpath='{.spec.openebsVolume.snapshotId}')
+rc=$?
+if [ $rc != 0 ]; then
+    echo "Failed to get snapshot name for volumesnapshot data: ${volume_snapshot_data_name}"
+    exit 1
+fi
+
+pv_name=$(kubectl get pvc cstor-vol1-1r-claim -o jsonpath='{.spec.volumeName}')
+rc=$?
+if [ $rc != 0 ]; then
+    echo "Failed to get PV name for PVC: cstor-vol1-1r-claim"
+    exit 1
+fi
+
+cvr_list=$(kubectl get cvr -n openebs -l openebs.io/persistent-volume=${pv_name} -o jsonpath='{.items[*].metadata.name}')
+rc=$?
+if [ $rc != 0 ]; then
+    echo "Failed to list CVRs of PV: ${pv_name}"
+    exit 1
+fi
+cvr_name=${cvr_list[0]}
+
+verify_snapshot_list_on_cvr "${cvr_name}" "openebs" "1" "${k8s_snapshot_name}"
+
+cstor_target_pod_list=$(kubectl get pod -n openebs -l openebs.io/persistent-volume=${pv_name},openebs.io/target=cstor-target -o jsonpath='{.items[*].metadata.name}')
+rc=$?
+if [ $rc != 0 ]; then
+    echo "Failed to list cStor target pods of PV: ${pv_name}"
+    exit 1
+fi
+cstor_target_pod_name=${cstor_target_pod_list[0]}
+
+snapshot_command=$(echo "kubectl exec -n openebs ${cstor_target_pod_name} -c cstor-istgt -- istgtcontrol snapcreate ${pv_name} istgt_snap1")
```

Comment:
why are we not using volumesnapshot to create snapshot?

Reply:
Just to not maintain any extra files to create snapshots.

Reply:
and another reason is we have to write more lines to fetch the volumesnapshotdata name to verify snapshot exists on CVR or not